mkosi/kmod.py

```diff
@@ -234,12 +238,27 @@ def process_kernel_modules(
                 p.rmdir()
 
         for fw in firmware:
+            # assume any files in /usr/lib/firmware/updates are required
+            if fw.is_relative_to(firmwared / "updates"):
+                continue
+
+            if fw in required and (root / fw).is_symlink():
+                # mopdules may depend on a symbolic link to the actual firmware file, so
+                # make sure those target files are included in the final image too.
+                p = (root / fw).resolve().relative_to(root)
+                required.add(p)
+                unneeded.discard(p)
+                continue
+
             if fw in required:
                 continue
```

Comment:
Since it tests the same thing again, this should probably be folded in, so that order between the blocks matters less.
```suggestion
            if fw in required
                if (root / fw).is_symlink():
                    # modules may depend on a symbolic link to the actual firmware file, so
                    # make sure those target files are included in the final image too.
                    p = (root / fw).resolve().relative_to(root)
                    required.add(p)
                    unneeded.discard(p)
                continue
```